### PR TITLE
BCDA-6291: Enhanced error handling: bcda-app /auth/token endpoint

### PR DESCRIPTION
**Pull request by ian-a-sawyer**
### Proposed Changes
- Added checks for custom errors during token retrieval
- Added table-driven tests for GetAuthToken endpoint handler
- Added table-driven tests for GetToken method in SSAS code

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation
- Tests pass locally
- Code is performant

### CODE REVIEW

**Review by krobertson3314 — CHANGES_REQUESTED** [review]

File: bcda/auth/api.go [hunk 1]
```diff
@@ -26,28 +28,39 @@ import (
...
 	}
 
 	// https://tools.ietf.org/html/rfc6749#section-5.1
 
-	body := []byte(fmt.Sprintf(`{"access_token": "%s", "expires_in": "%s", "token_type":"bearer"}`, token, expiresIn))
 	w.Header().Set("Content-Type", "application/json")
 	w.Header().Set("Cache-Control", "no-store")
 	w.Header().Set("Pragma", "no-cache")
-	_, err = w.Write(body)
+	_, err = w.Write([]byte(tokenInfo))
```
Comment (krobertson3314) [inline bcda/auth/api.go:60]: Is tokenInfo in same structure as before for the end user? 

{"access_token": "%s", "expires_in": "%s", "token_type":"bearer"}`

Comment (ian-a-sawyer) [reply bcda/auth/api.go:60]: Yes.

File: bcda/auth/api.go [hunk 2]
```diff
@@ -26,28 +28,39 @@ import (
...
+	tokenInfo, err := GetProvider().MakeAccessToken(Credentials{ClientID: clientId, ClientSecret: secret})
 	if err != nil {
-		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		switch err.(type) {
+		case *customErrors.RequestTimeoutError:
+			//default retrySeconds: 1 second (may convert to environmental variable later)
+			retrySeconds := strconv.FormatInt(int64(1), 10)
+			w.Header().Set("Retry-After", retrySeconds)
+			http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		case *customErrors.UnexpectedSSASError:
```
Comment (krobertson3314) [inline bcda/auth/api.go:47]: Lets add *customErrors.InternalParsingError to also be here & write InternalServerError (not let it go to default 401 unauthorized)

File: bcda/auth/api_test.go [hunk 3]
```diff
@@ -45,56 +47,106 @@ func (s *AuthAPITestSuite) SetupSuite() {
...
+	client := s.server.Client()
+
+	tests := []struct {
+		ScenarioName          string
+		ErrorToReturn         error
+		StatusCode            int
+		HeaderRetryAfterValue string
+	}{
+		{"Token Request Timeout Error Return 503", &customErrors.RequestTimeoutError{Err: errors.New(errorHappened), Msg: errMsg}, 503, "1"},
+		{"Token Unexpected SSAS Error Return 500", &customErrors.UnexpectedSSASError{Err: errors.New(errorHappened), Msg: errMsg}, 500, constants.EmptyString},
```
Comment (krobertson3314) [inline bcda/auth/api_test.go:74]: add one more scenario for InternalParsingError is returned during MakeAccessToken, and verify 500 is the status code

**Review by krobertson3314 — APPROVED** [review]

**Pull request merged**